bare-arm/Makefile

```diff
@@ -22,15 +31,23 @@ else
 CFLAGS += -Os -DNDEBUG
 endif
 
+ifeq ($(CROSS), 1)
 LDFLAGS = -nostdlib -T stm32f405.ld -Map=$@.map --cref
+else
+LD = gcc
+LDFLAGS = -m32 -Wl,-Map=$@.map,--cref
+endif
 LIBS =
 
 SRC_C = \
 	main.c \
-#	printf.c \
-	string0.c \
-	malloc0.c \
-	gccollect.c \
+	../stmhal/printf.c \
```

Comment:
You don't need the "../".  With it, it creates stmhal/ directory within bare-arm to store .o files.  See esp8266 Makefile.